ipa-core/src/ff/ec_prime_field.rs

```diff
@@ -203,6 +244,31 @@ impl FromRandom for Fp25519 {
     }
 }
 
+macro_rules! impl_share_from_random {
+    ($width:expr) => {
+        impl FromPrss for AdditiveShare<Fp25519, $width> {
+            fn from_prss_with<P: SharedRandomness + ?Sized, I: Into<PrssIndex>>(
+                prss: &P,
+                index: I,
+                _params: (),
+            ) -> AdditiveShare<Fp25519, $width> {
+                let (l_arr, r_arr) = StdArray::from_tuple_iter(
+                    prss.generate_chunks_iter::<_, U2>(index)
+                        .map(|(l_rand, r_rand)| {
+                            (Fp25519::from_random(l_rand), Fp25519::from_random(r_rand))
+                        })
+                        .take($width),
+                );
+                AdditiveShare::new_arr(l_arr, r_arr)
+            }
+        }
+    };
+}
+
+impl_share_from_random!(16);
```

Comment:
I am in favour of deleting unused implementations or covering them with tests

Reply:
I scrubbed all of the impls and removed everything that is unused.